numpy/core/src/umath/ufunc_object.c

```diff
@@ -5284,11 +5301,14 @@ ufunc_dealloc(PyUFuncObject *ufunc)
     PyArray_free(ufunc->ptr);
     PyArray_free(ufunc->op_flags);
     Py_XDECREF(ufunc->userloops);
-    Py_XDECREF(ufunc->obj);
     if (ufunc->identity == PyUFunc_IdentityValue) {
         Py_DECREF(ufunc->identity_value);
     }
-    PyArray_free(ufunc);
+    if (ufunc->obj != NULL) {
+        PyObject_GC_UnTrack((PyObject *)ufunc);
+        Py_DECREF(ufunc->obj);
```

Comment:
Can you convince me that the order of:
* `Py_DECREF(ufunc->identity_value)`
* `PyObject_GC_UnTrack((PyObject *)ufunc)`
* `Py_DECREF(ufunc->obj)`

is correct? Putting the untracking between decref seems asymmetric to me.

Reply:
The [documentation](https://docs.python.org/3/c-api/gcsupport.html#c._PyObject_GC_TRACK) supports your claim. Refactoring.